packages/legend-studio/src/models/metamodels/pure/model/packageableElements/mapping/xStore/XStorePropertyMapping.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2020 Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { hashArray } from '@finos/legend-studio-shared';
+import type { Hashable } from '@finos/legend-studio-shared';
+import { CORE_HASH_STRUCTURE } from '../../../../../../MetaModelConst';
+import type { RawLambda } from '../../../rawValueSpecification/RawLambda';
+import type { Stubable } from '../../../Stubable';
+import type { PropertyMappingVisitor } from '../PropertyMapping';
+import { PropertyMapping } from '../PropertyMapping';
+
+export class XStorePropertyMapping
+  extends PropertyMapping
+  implements Hashable, Stubable {
+  crossExpression!: RawLambda;
+
+  get hashCode(): string {
+    return hashArray([
+      CORE_HASH_STRUCTURE.XSTORE_PROPERTY_MAPPING,
+      super.hashCode,
+      this.crossExpression,
+    ]);
+  }
+
+  get isStub(): boolean {
```

Comment:
do we use this? if not, can we get rid of it, I think we're about to try to deprecate this annoying `Stub` interface?

Reply:
We need to implement this as we check for stub when transforming to v1. It makes no sense not to implement this here unless we change this on all property mappings and remove the isstub  on all of them when doing the transforming which requires a separate discussion and separate PR.